frontend/src/components/core/Sidebar/SidebarNav.tsx

```diff
@@ -76,12 +87,16 @@ const SidebarNav = ({
   )
 
   const toggleExpanded = useCallback(() => {
-    if (!expanded && isOverflowing) {
+    if (!expanded) {
       setExpanded(true)
+      setPagesToShow(-1)
+      localStorage.setItem("navExpanded", "true")
     } else if (expanded) {
```

Comment:
Agree with CodeQL here, since there are only 2 states for `expanded` boolean, an if/else would be sufficient:
```javascript
if (!expanded) {
      setExpanded(true)
      setPagesToShow(-1)
      localStorage.setItem("navExpanded", "true")
    } else {
      setExpanded(false)
      setPagesToShow(6)
      localStorage.setItem("navExpanded", "false")
    }
  }, [expanded])
```

Reply:
Makes sense! Updated in [6919ec3](https://github.com/streamlit/streamlit/pull/6050/commits/6919ec3e85adc6a8573561634ed007127edb35e1)